d2/src/main/java/com/linkedin/d2/balancer/config/CanaryDistributionStrategyConverter.java

```diff
@@ -0,0 +1,135 @@
+          config.setStrategy(StrategyType.DISABLED);
+          break;
+        }
+
+        PercentageStrategyProperties toPercentageProperties = new PercentageStrategyProperties();
+        toPercentageProperties.setScope(coerce(fromPercentageProperties.get(PropertyKeys.PERCENTAGE_SCOPE), Double.class));
+        config.setPercentageStrategyProperties(toPercentageProperties);
+        break;
+      case TARGET_HOSTS:
+        Map<String, Object> fromTargetHostsProperties = properties.getTargetHostsStrategyProperties();
+        if (fromTargetHostsProperties == null || !fromTargetHostsProperties.containsKey(PropertyKeys.TARGET_HOSTS))
+        {
+          LOG.warn("No properties set for targetHosts strategy in CanaryDistributionStrategy: " + fromTargetHostsProperties + ". Fall back to DISABLED.");
+          config.setStrategy(StrategyType.DISABLED);
+          break;
+        }
+
+        List<String> hosts = (List<String>) fromTargetHostsProperties.get(PropertyKeys.TARGET_HOSTS);
+        TargetHostsStrategyProperties toTargetHostsProperties = new TargetHostsStrategyProperties();
+        toTargetHostsProperties.setTargetHosts(new StringArray(hosts));
+        config.setTargetHostsStrategyProperties(toTargetHostsProperties);
+        break;
+      case TARGET_APPLICATIONS:
+        Map<String, Object> fromTargetAppsProperties = properties.getTargetApplicationsStrategyProperties();
+        if (fromTargetAppsProperties == null
+                || !fromTargetAppsProperties.containsKey(PropertyKeys.TARGET_APPLICATIONS)
+                || !fromTargetAppsProperties.containsKey(PropertyKeys.PERCENTAGE_SCOPE))
+        {
+          LOG.warn("No properties set for targetApplications strategy in CanaryDistributionStrategy: " + fromTargetAppsProperties + ". Fall back to DISABLED.");
+          config.setStrategy(StrategyType.DISABLED);
+          break;
+        }
+
+        List<String> apps = (List<String>) fromTargetAppsProperties.get(PropertyKeys.TARGET_APPLICATIONS);
+        TargetApplicationsStrategyProperties toTargetAppsProperties = new TargetApplicationsStrategyProperties();
+        toTargetAppsProperties.setTargetApplications(new StringArray(apps));
+        toTargetAppsProperties.setScope(coerce(fromTargetAppsProperties.get(PropertyKeys.PERCENTAGE_SCOPE), Double.class));
+        config.setTargetApplicationsStrategyProperties(toTargetAppsProperties);
+        break;
+      case DISABLED:
+        break;
+      default:
+        throw new IllegalStateException("Unexpected value: " + type);
+    }
+    return config;
+  }
+
+  public static CanaryDistributionStrategy toProperties(D2CanaryDistributionStrategy config)
+  {
+    Map<String, Object> percentageStrategyProperties = new HashMap<>();
+    Map<String, Object> targetHostsStrategyProperties = new HashMap<>();
+    Map<String, Object> targetApplicationsStrategyProperties = new HashMap<>();
+
+    String strategyName;
+    switch(config.getStrategy()) {
```

Comment:
Let's follow the convention in pegasus to have brackets on the new line. Same for all occurrences below